src/main/ruby/truffleruby/core/io.rb

```diff
@@ -2158,6 +2158,23 @@ def set_encoding(external, internal=nil, options=undefined)
     self
   end
 
+  def set_encoding_by_bom
+    unless binmode?
+      raise ArgumentError, 'ASCII incompatible encoding needs binmode'
+    end
+
+    if !Primitive.nil?(internal_encoding)
+      raise ArgumentError, 'encoding conversion is set'
+    end
+
+    if !Primitive.nil?(external_encoding) && external_encoding != Encoding::ASCII_8BIT
+      raise ArgumentError, "encoding is set to #{external_encoding} already"
+    end
+
+    external = strip_bom
+    external && @external = Encoding.find(external)
```

Comment:
I find this line super hard to read and understand - can we break it down?

```ruby
external = strip_bom
unless external
  @external = Encoding.find(external)
end
```

Is it that? I still can't really wrap my head around that either to be honest.

Reply:
It's `if external` actually, I think in `if` form it's clear

Reply:
Yes, agreed. That was too much code-golf 😅 Changed as suggested.